wp-sentry.php

```diff
@@ -10,15 +10,22 @@
  * License: MIT
  */
 
-// Exit if accessed directly.
+// Exit if accessed directly
 defined( 'ABSPATH' ) || exit;
 
-// If the plugin was already loaded as a mu-plugin do not load again.
-if ( defined( 'WP_SENTRY_MU_LOADED' ) ) {
+// If the plugin was already loaded as a mu-plugin or from somewhere else do not load again
+if ( defined( 'WP_SENTRY_MU_LOADED' ) || defined( 'WP_SENTRY_LOADED' ) ) {
 	return;
 }
 
-// Make sure the PHP version is at least 7.2.
+define( 'WP_SENTRY_LOADED', true );
+
+// Load the WordPress plugin API early so hooks can be used even if Sentry is loaded before WordPress
+if ( ! function_exists( 'add_action' ) ) {
+	require_once ABSPATH . '/wp-includes/plugin.php';
+}
+
+// Make sure the PHP version is at least 7.2
 if ( ! defined( 'PHP_VERSION_ID' ) || PHP_VERSION_ID < 70200 ) {
 	if ( is_admin() ) {
```

Comment:
This is a rare case but the use of `is_admin()` here may cause a fatal if someone is including the file in wp-config.php and not using PHP 7.2. I think we can remove the check here and rely fully on the `admin_notices` hook which is only used in the admin.

Probably good to add the `Requires PHP` file header too, see https://developer.wordpress.org/plugins/plugin-basics/header-requirements/#header-fields.

Reply:
You are right and I agree 👍 good call!